gtda/diagrams/features.py

```diff
@@ -701,3 +707,150 @@ def transform(self, X, y=None):
                                         self.n_bins, self.n_bins).\
             transpose((1, 0, 2, 3))
         return Xt
+
+
+@adapt_fit_transform_docs
+class Silhouette(BaseEstimator, TransformerMixin):
+    """Silhouettes of persistence diagrams.
+
+    Based on ideas in [1]_. Given a persistence diagram consisting of
+    birth-death-dimension triples [b, d, q], subdiagrams corresponding to
+    distinct homology dimensions are considered separately.
+    Let :math:`D=\\{p_k/\\}_{k=1}^N` be a subdiagram, where
```

Comment:
We should be consistent with the other transformers in this module and probably create a glossary entry for this and removing the mathematical definition from the docstring. I'll create an issue for the theory page.